main.ts

```diff
@@ -217,16 +217,32 @@ class Translator {
         break;
 
       case ts.SyntaxKind.VariableDeclarationList:
+        // Note that a VariableDeclarationList can only occur as part of a for loop.
         var varDeclList = <ts.VariableDeclarationList>node;
-        this.visitList(varDeclList.declarations, ';');
+        this.visitList(varDeclList.declarations, ',');
         break;
 
       case ts.SyntaxKind.VariableDeclaration:
         var varDecl = <ts.VariableDeclaration>node;
-        if (varDecl.type) {
-          this.visit(varDecl.type);
-        } else {
-          this.emit('var');
+        /*
+         * All variables in a variable declaration list must have the same type. Since
+         * we are doing syntax directed translation, we cannot reliably determine if distinct
+         * variables are declared with the same type or not. Hence we support the following cases:
+         * - A variable declaration list with a single variable can be explicitly typed.
+         * - When more than one variable is in the list, all must be implicitly typed.
+         */
+        var firstDecl = varDecl.parent.declarations[0];
+        var msg = 'Variables in a declaration list of more than one variable cannot by typed';
+        if (firstDecl === varDecl) {
+          if (!varDecl.type) {
```

Comment:
Could you extract this to a method "visitVarDeclType"?

Then you could also use early exits from the function to make the boolean conditions a bit easier to read.